Lib/sre_compile.py

```diff
@@ -333,10 +333,7 @@ def _optimize_charset(charset, iscased=None, fixup=None, fixes=None):
     # compress character map
     runs = []
     q = 0
-    while True:
-        p = charmap.find(1, q)
-        if p < 0:
-            break
+    while (p := charmap.find(1, q)) >= 0:
```

Comment:
bug (I think): changed `<` to `>=`

Reply:
The condition in the while is always *not* the break condition.  So in most of the others, the binary `if not data` becomes `while data`.

Reply:
Oh yeah, my bad! I just misread this one.